src/main/java/com/project/parkinglot/service/parking_area/impl/ParkingAreaCreateServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.project.parkinglot.service.parking_area.impl;
+
+import com.project.parkinglot.exception.parkingarea.ParkingAreAlreadyExistException;
+import com.project.parkinglot.model.ParkingArea;
+import com.project.parkinglot.model.dto.request.parking_area.ParkingAreaCreateRequest;
+import com.project.parkinglot.model.entity.ParkingAreaEntity;
+import com.project.parkinglot.model.mapper.parking_area.ParkingAreaCreateRequestToParkingAreaEntityMapper;
+import com.project.parkinglot.model.mapper.parking_area.ParkingAreaEntityToParkingAreaDomainModelMapper;
+import com.project.parkinglot.repository.ParkingAreaRepository;
+import com.project.parkinglot.service.parking_area.ParkingAreaCreateService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ParkingAreaCreateServiceImpl implements ParkingAreaCreateService {
+
+    private final ParkingAreaRepository parkingAreaRepository;
+
+    private final ParkingAreaCreateRequestToParkingAreaEntityMapper parkingAreaCreateRequestToParkingAreaEntityMapper =
+            ParkingAreaCreateRequestToParkingAreaEntityMapper.initialize();
+
+    private final ParkingAreaEntityToParkingAreaDomainModelMapper parkingAreaEntityToParkingAreaDomainModelMapper =
+            ParkingAreaEntityToParkingAreaDomainModelMapper.initialize();
+
+    @Override
+    public ParkingArea createParkingArea(
+            final ParkingAreaCreateRequest parkingAreaCreateRequest
+    ) {
+        this.checkParkingAreaNameAndLocationUniqueness(
+                parkingAreaCreateRequest.getName(),
+                parkingAreaCreateRequest.getLocation()
+        );
+
+        ParkingAreaEntity parkingAreaToBeCreate = parkingAreaCreateRequestToParkingAreaEntityMapper
+                .map(parkingAreaCreateRequest);
+
+        parkingAreaRepository.save(parkingAreaToBeCreate);
+
+        return parkingAreaEntityToParkingAreaDomainModelMapper
+                .map(parkingAreaToBeCreate);
+    }
+
+    private void checkParkingAreaNameAndLocationUniqueness(
```

Comment:
GlobalExceptionHandler sınıfında şuna benzer bir method oluşturabilirmisin
```

@ExceptionHandler(ParkingAreAlreadyExistException.class)
    public ResponseEntity<String> handleParkingAreAlreadyExistException(ParkingAreAlreadyExistException e) {
        return ResponseEntity.status(HttpStatus.CONFLICT).body("Parking area already exists");
    }
    
```

Bunun çalışıp çalışılmadığını öğrenmek için Exception testi yazmak gerekiyor.

Reply:
Kodu GlobalExceptionHandler sınıfına ekleyeceğim

Reply:
401 hatası alıyorduk. Bunu yazarsak o sorunu çözeceğimizi düşünüyorum.